dspace-api/src/test/java/org/dspace/service/impl/HttpConnectionPoolServiceTest.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.service.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.dspace.AbstractDSpaceTest;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockserver.client.MockServerClient;
+import org.mockserver.junit.MockServerRule;
+
+/**
+ *
+ * @author Mark H. Wood <mwood@iupui.edu>
+ */
+public class HttpConnectionPoolServiceTest
+        extends AbstractDSpaceTest {
+    private static ConfigurationService configurationService;
+
+    @Rule
+    public MockServerRule mockServerRule = new MockServerRule(this);
+
+    private MockServerClient mockServerClient;
+
+    @BeforeClass
+    public static void initClass() {
+        configurationService = DSpaceServicesFactory.getInstance()
+                .getConfigurationService();
+    }
+
+    /**
+     * Test of getClient method, of class HttpConnectionPoolService.
+     * @throws java.io.IOException if a connection cannot be closed.
+     * @throws java.net.URISyntaxException when an invalid URI is constructed.
+     */
+    @Test
+    public void testGetClient()
```

Comment:
This is a good basic "smoke test". I wonder if it'd be possible to add a few other tests...for example.

* Set `solr.client.maxTotalConnections` to 1 and verify two calls to getClient() end up with the same connection.
* Set `solr.client.keepAlive` to a small value, and verify the connection auto-closes after that time.
* Set `solr.client.timeToLive` to a small value, and verify connection ejected from pool

These are just some ideas of ways we might add a few more additional tests here. But, if any of these seem really hard to easily verify in code, I'm OK with moving forward with this PR with minimal tests.

Reply:
@mwoodiupui : Previously I had asked if it'd be possible to add a few more tests here to verify each of the basic `solr.client.*` configs are working as expected?  See https://github.com/DSpace/DSpace/pull/7987/files#r766220588  As noted there, if these tests seem too difficult to build, let me know.  We can move this PR forward without these tests, but I wanted to ask whether these tests would be possible to add.

Reply:
There may not be a good way to do this.  The ClientBuilder uses its configured ConnectionManager to get a connection.  There seems no good way to get at the actual connection to see if it is an identical reference to another client's connection.  The actual pool is hidden inside ConnectionManager and well insulated from the outside world.

Reply:
@mwoodiupui : Ok, we can leave it as is then.  I was just trying to see what was possible & I trust your judgment here. I'll add this to my list to re-review since it looks like you've updated it today.